pontoon/insights/tasks.py

```diff
@@ -0,0 +1,391 @@
+        .distinct()
+    )
+
+    return group_dict_by(actions, "translation__locale")
+
+
+def get_suggestions():
+    """Get currently unreviewed suggestions."""
+    suggestions = Translation.objects.filter(
+        approved=False,
+        fuzzy=False,
+        rejected=False,
+        entity__obsolete=False,
+        entity__resource__project__disabled=False,
+        entity__resource__project__system_project=False,
+        entity__resource__project__visibility="public",
+    ).values("locale", "date")
+
+    return group_dict_by(suggestions, "locale")
+
+
+def get_activity_actions(start_of_today):
+    """Get actions of the previous day, needed for the Translation and Review activity charts."""
+    actions = (
+        ActionLog.objects.filter(
+            created_at__gte=start_of_today - timedelta(days=1),
+            created_at__lt=start_of_today,
+            translation__entity__resource__project__system_project=False,
+            translation__entity__resource__project__visibility="public",
+        )
+        .prefetch_related("translation__errors")
+        .prefetch_related("translation__warnings")
+        .values(
+            "action_type",
+            "translation",
+            "translation__locale",
+            "translation__approved",
+            "translation__fuzzy",
+            "translation__rejected",
+            "translation__errors",
+            "translation__warnings",
+            "translation__machinery_sources",
+            "translation__user",
+            "translation__approved_user",
+        )
+    )
+
+    return group_dict_by(actions, "translation__locale")
+
+
+def get_entities(start_of_today):
+    """Get entities created on the previous day."""
+    entities = Entity.objects.filter(
+        date_created__gte=start_of_today - timedelta(days=1),
+        date_created__lt=start_of_today,
+        obsolete=False,
+        resource__project__disabled=False,
+        resource__project__system_project=False,
+        resource__project__visibility="public",
+    ).values("pk", "resource__translatedresources__locale")
+
+    return group_dict_by(entities, "resource__translatedresources__locale")
+
+
+def get_locale_insights_snapshot(
+    locale,
+    start_of_today,
+    privileged_users,
+    contributors,
+    active_users_actions,
+    suggestions,
+    activity_actions,
+    entities,
+):
+    """Create LocaleInsightsSnapshot instance for the given locale and day using given data."""
+    all_managers, all_reviewers = get_all_users_data(privileged_users)
+    all_contributors = {c["user"] for c in contributors}
+
+    total_managers = len(all_managers)
+    total_reviewers = len(all_reviewers)
+    total_contributors = len(all_contributors)
+
+    active_users_last_month = get_active_users_data(
+        start_of_today,
+        privileged_users,
+        active_users_actions,
+        all_managers,
+        all_reviewers,
+        days=365 / 12,
```

Comment:
I'm not sure about the required accuracy of the Insights dashboard, but the number of days is going to be wrong for leap years. Also, the number of days in a month changes too. The same comment applies to the calculation above.

Reply:
I'll see if I can use `relativedelta` instead.